### DISCUSSION TITLE
Changing color for identifiers

### DISCUSSION THREAD

**Discussion by bettydots** [tiagovla/tokyodark.nvim#27] [Q&A] [answered]
First off I'd like to say I love this theme. It's wonderful for Markdown reading and generally very pleasing

That being said I don't love that the variable names are orange. Makes everything looks very orange and I would like to change them to white so I have an orange "let" keyword and a white variable name, for example.

I have been racking my brain trying to figure out how to affect the settings to do this so I decided to ask for help, because the highlight group stuff in Nvim is wrecking my brain. Could anyone help? Also, any info y'all might have as to how to edit certain highlights in a custom colorscheme theme would be greatly appreciated, because I cannot understand it no matter how hard I try

here's what I tried, probably incredibly wrong haha
```
{
    "tiagovla/tokyodark.nvim",
    opts = {
        transparent_background = true, -- set background to transparent
        custom_highlights = {
          Identifiers = {
            guifg = "#FFFFFF",
          },
        },
        -- custom options here
    },
    config = function(_, opts)
        require("tokyodark").setup(opts) -- calling setup is optional
        vim.cmd [[colorscheme tokyodark]]
    end,
    },
```

EDIT: I would like to update my question because in my testing I tried to set the style for my identifiers to not be italic and this is also not working correctly, despite me having the same syntax as the setup page (at least as far as I can tell). Here is my attempt to do so. Can't really tell what it is I'm doing wrong so I'd definitely appreciate some help!

```
{
      "tiagovla/tokyodark.nvim",
      opts = {
        transparent_background = true, -- set background to transparent
        styles = {
          identifiers = { italic = false },
        },
      },
      config = function(_, opts)
        require("tokyodark").setup(opts)   -- calling setup is optional
        vim.cmd [[colorscheme tokyodark]]
      end,
},
```

**Comment by bettydots** [comment]
Ok, I made a little progress as to my edit. I think the styles for the identifiers weren't working because I had to include all the options whether I needed to set them or not. This got rid of the italics for the "Identifiers"

```
{
      "tiagovla/tokyodark.nvim",
      opts = {
        transparent_background = true, -- set background to transparent
        gamma = 1.00,
        styles = {
          comments = { italic = true }, -- style for comments
          keywords = { italic = true }, -- style for keywords
          identifiers = { italic = false }, -- style for identifiers
          functions = {}, -- style for functions
          variables = {}, -- style for variables
        },
        custom_highlights = {} or function(highlights, palette) return {} end, -- extend highlights
        custom_palette = {} or function(palette) return {} end, -- extend palette
        terminal_colors = true, -- enable terminal colors
       },
      config = function(_, opts)
        require("tokyodark").setup(opts)   -- calling setup is optional
        vim.cmd [[colorscheme tokyodark]]
      end,
},
```

What would I need to do now in order to make the Identifiers be white instead of orange? I apologize for my lack of expertise also. My coding journey is just starting and though I try to look up most things myself this particular problem is confusing me beyond belief

**Comment by bettydots** [comment]
Ok, I was finally able to change the configuration to make the Identifier have the same palette as the fg. Here's what I did, feel free to let me know if there is a better way! I had to look through the palette.lua and highlights.lua files to find the values I wanted. Perhaps there is a way to incorporate these config nuances in the readme?

Anyway, Thank you for the theme. I'll close the discussion with this one

```
    {
      "tiagovla/tokyodark.nvim",
      opts = {
        transparent_background = true, -- set background to transparent
        gamma = 1.00,
        styles = {
          comments = { italic = true }, -- style for comments
          keywords = { italic = true }, -- style for keywords
          identifiers = { italic = false }, -- style for identifiers
          functions = {}, -- style for functions
          variables = {}, -- style for variables
        },
        custom_highlights = {
          Identifier = { fg = "fg" },
        } or function(highlights, palette) return {} end, -- extend highlights
        custom_palette = {} or function(palette) return {} end, -- extend palette
        terminal_colors = true, -- enable terminal colors
       },
      config = function(_, opts)
        require("tokyodark").setup(opts)   -- calling setup is optional
        vim.cmd [[colorscheme tokyodark]]
      end,
    },

```

**Reply by tiagovla** [reply, marked as answer]
Hello,

Maybe something like this should work:
```lua
return {
    "tiagovla/tokyodark.nvim",
    opts = {
        custom_highlights = function(hl, p)
            hl["Identifier"] = { fg = p.fg }
            return {}
        end,
    },
    config = function(_, opts)
        require("tokyodark").setup(opts)
        require("tokyodark").colorscheme()
    end,
}
```

The colors available are [here](https://github.com/tiagovla/tokyodark.nvim/blob/4bfb42924274abc5de9f5f4779075b77c6112c85/lua/tokyodark/palette.lua#L5-L31) and the highlight groups [here](https://github.com/tiagovla/tokyodark.nvim/blob/4bfb42924274abc5de9f5f4779075b77c6112c85/lua/tokyodark/highlights.lua#L112C57-L112C57). The `hl` inside the `custom_highlights` function exposes the full highlights table and `p` the colors.

Also, returning a table will merge it to the current `highlights` one. So this is also possible:
```lua
---
custom_highlights = function(hl, p)
    return {
        Identifier = { fg = p.fg },
    }
end,
---
```

You can check the highlight groups by placing the cursor over a word and using the command `:Inspect`. If you have a screenshot showing the highlights you want to replace for a given language, that would be easier to help you.

**Reply by tiagovla** [reply]
Returning a table like you did also extends it.
```lua
---
custom_highlights = {
    Identifier = { link = "Fg" }, -- or { fg = "#000000" }
}
---
```
So there are three ways of doing it.

**Reply by bettydots** [reply]
Thank you, I was having trouble understanding how to set up the function. Your first solution worked like a charm and helped me understand the whole setup process better. Here's my setup with white variable names, as well as different color headers in Markdown.

```
 {
      "tiagovla/tokyodark.nvim",
      opts = {
        transparent_background = true, -- set background to transparent
        gamma = 1.00,
        styles = {
          comments = { italic = true }, -- style for comments
          keywords = { italic = false }, -- style for keywords
          identifiers = { italic = false }, -- style for identifiers
          functions = {}, -- style for functions
          variables = {}, -- style for variables
        },
        custom_highlights = function(h1,p)
          h1["Identifier"] = { fg = p.fg }
          h1["markdownH2"] = { fg = p.orange }
          h1["markdownH3"] = { fg = p.yellow }
          h1["markdownH4"] = { fg = p.blue }
          h1["markdownH5"] = { fg = p.cyan }
          h1["markdownH6"] = { fg = p.purple }
          h1["markdownCode"] = { fg = p.green }
          h1["markdownCodeBlock"] = { fg = p.green }
          return{}
        end,
        custom_palette = {} or function(palette) return {} end, -- extend palette
        terminal_colors = true, -- enable terminal colors
       },
      config = function(_, opts)
        require("tokyodark").setup(opts)   -- calling setup is optional
        vim.cmd [[colorscheme tokyodark]]
      end,
    },
```

Thanks again for the help!
